parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreBase.java

```diff
@@ -231,7 +231,9 @@ private void sizeCheck() {
       long usedMem = writer.getCurrentPageBufferedSize();
       long rows = rowCount - writer.getRowsWrittenSoFar();
       long remainingMem = props.getPageSizeThreshold() - usedMem;
-      if (remainingMem <= thresholdTolerance || rows >= pageRowCountLimit) {
+      if (remainingMem <= thresholdTolerance ||
+          rows >= pageRowCountLimit ||
+          writer.getValueCount() >= Integer.MAX_VALUE / 2) {
```

Comment:
Do you think it worths being added to the configuration properties for more flexibility?

Reply:
Added :) 
Let me know if it looks okay